Documentation/teaching/lectures/interrupts.rst

```diff
@@ -0,0 +1,558 @@
+    +------------------------------+-+--+----+--+-------------------+
+    |                              |                                |
+    |       segment selector       |        offset (0..15)          |
+    |                              |                                |
+    +------------------------------+--------------------------------+
+     31                             15                             0
+
+
+Interrupt handler address
+-------------------------
+
+In order to find the interrupt handler address we first need to find the start
+address of the code segment where interrupt handler resides. For this we
+use the segment selector to index into GDT/LDT where we can find the corresponding
+segment descriptor. This will provide the start address kept in the 'base' field.
+Using base address and the offset we can now go at the start of the interrupt handler.
+
+
+.. slide:: Interrupt handler address
+   :inline-contents: True
+   :level: 2
+
+   .. ditaa::
+
+     
+                      Interrupt Descriptor
+          +----------------------------------------------+
+          |                                              |
+          |  +------------------+  +--------+  +------+  |
+          |  | segment selector |  |  offset|  |  PL  |  |
+          |  +----+-------------+  +---+----+  +------+  |
+          |       |                    |                 |
+          +----------------------------------------------+
+                  |                    |
+                  |                    |
+    +-------------+                    +---------------------------->  +---------------+
+    |                                                               ^  |  ISR address  |
+    |                   Segment Descriptor                          |  +---------------+
+    |     +----------------------------------------------+          |
+    |     |                                              |          |
+    +---->|  +------------------+  +--------+  +------+  |          |
+          |  |      base        |  |  limit |  |  PL  |  |          |
+          |  +---------+--------+  +--------+  +------+  |          |
+          |            |                                 |          |
+          +----------------------------------------------+          |
+                       |                                            |
+                       +--------------------------------------------+
+
+
+Stack of interrupt handler
+--------------------------
+
+Similar with control transfer to a normal function, a control transfer
+to an interrupt or exception handler uses the stack to store the 
+information needed for returning to the interrupted code.
+
+As can be seen in the figure below, an interrupt pushes the EFLAGS register
+before saving the address of the interrupted instruction. Certain types
+of exceptions also cause an error code to be pushed on the stack to help
+debug the exception.
+
+
```

Comment:
I think it is worth talking about what happens in the case of a privilege transition. In the figure shouldn't it be NEW SS:ESP from TSS (and not OLD SS:EPS from TSS)? 

Maybe redraw the figure to make it clear that the privilege transition changes SS:ESP.

Reply:
True. Will try to make a better image.

I got it form here, where there are no much explanations.

https://css.csail.mit.edu/6.858/2015/readings/i386.pdf